docs/open-source/external-action.mdx

```diff
@@ -0,0 +1,265 @@
+
+In the [Meeting Assistant Example](/open-source/external-action#meeting-assistant-example) below, the user’s API could return back a JSON response that looks like:
+
+```json
+{
+  "result": {
+    "success": true
+  },
+  "agent_message": "I've set up a calendar appointment at 10:30am tomorrow for 30 minutes"
+}
+```
+
+## EA Local Response Server Example:
+
+The following is an example of a quick start to enable testing external actions locally.
+
+Running `fastapi dev app.py` will run the server @ `http://127.0.0.1:8000` and can be used for external actions locally!
+
+<CodeGroup>
+```python app.py
+import asyncio
+import base64
+import hashlib
+import hmac
+import json
+import time
+
+from fastapi import FastAPI, HTTPException, Request, status
+from pydantic import BaseModel
+
+app = FastAPI()
+
+action_secret = ""
+
+
+def is_signature_valid(request: Request, json_payload: dict):
+    signature_secret_as_bytes = base64.b64decode(action_secret)
+    decoded_digest = base64.b64decode(request.headers.get("x-vocode-signature"))
+    calculated_digest = hmac.new(
+        signature_secret_as_bytes, json.dumps(json_payload).encode(), hashlib.sha256
+    ).digest()
+    return hmac.compare_digest(decoded_digest, calculated_digest)
+
+
+@app.post("/external_action")
+def update_item(request: Request):
+    json_payload = asyncio.run(request.json())
+    print(json_payload)
+    if not is_signature_valid(request, json_payload):
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Signature verification failed for payload",
+        )
+    time.sleep(3)
+    return {
+        "result": {"success": True},
+    }
+````
+```txt requirements.txt
+pydantic==2.*
+fastapi==0.111.*
+````
+
+</CodeGroup>
+
+## Meeting Assistant Example:
+
+This is an example of how to configure a Meeting Assistant Action which will attempt to book a meeting for 30 minutes or an hour at any time ending in a zero (ie 10:30am is okay but 10:35am is not)
+
+```python Python
+import json
+import base64
+import os
+
+from vocode.streaming.action.execute_external_action import (
+    ExecuteExternalAction,
+    ExecuteExternalActionVocodeActionConfig,
+)
+
+ACTION_INPUT_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "length": {
+            "type": "string",
+            "enum": ["30m", "1hr"],
+        },
+        "time": {
+            "type": "string",
+            "pattern": "^\d{2}:\d0[ap]m$",
+        },
+    },
+}
+
+action_config = {
+    "name": "Meeting_Booking_Assistant",
+    "description": "Book a meeting for a 30 minute or 1 hour call.",
+    "url": "http://example.com/booking",
+    "speak_on_send": True,
+    "speak_on_receive": True,
+    "input_schema": json.dumps(ACTION_INPUT_SCHEMA),
+    "signature_secret": base64.b64encode(os.urandom(32)).decode(),
+}
+
+action = ExecuteExternalAction(
```

Comment:
can we link the Action Agents doc here? this isn't how you would use the action

also - in this PR can we update the the Action Agents or the Agent Factories doc to actually show how to insert a user created action into a conversation? both of those docs are missing that